partiql-parser/src/lalr/lexer.rs

```diff
@@ -31,9 +136,16 @@ pub enum LexicalError {
 type CommentToken = SpannedResult<String, usize, LexicalError>;
 
 #[derive(Logos, Debug, Clone, PartialEq, Eq)]
+#[logos(extras = &'s mut LineOffsetTracker)]
```

Comment:
qq: what does this `logos` annotation w/ `extras` do?

Reply:
> Associated type Extras for the particular lexer. This can be set using #[logos(extras = MyExtras)] and accessed inside callbacks.

From https://docs.rs/logos/0.12.0/logos/trait.Logos.html#associatedtype.Extras